traitsui/testing/tester/qt4/implementation/list_editor.py

```diff
@@ -64,49 +43,25 @@ def register(cls, registry):
         registry : TargetRegistry
             The registry being registered to.
         """
-        registry.register_solver(
-            target_class=NotebookEditor,
-            locator_class=locator.Index,
-            solver=cls.from_location,
-        )
-
+        super().register(registry)
         register_nested_ui_solvers(
```

Comment:
The `solvers` list in `_BaseSourceWithLocation` was removed in #1191 because it wasn't used then. Now we have a use case for it. Would you like to add it back? Then we don't need to worry about overriding the base class method.

Reply:
That makes sense to me, but then it is a bit awkward as we wouldn't be able to use `register_nested_ui_solvers` as it is currently written.  Instead we would need to set something like:
```
solver = [(locator.NestedUI, solver ...) , (locator.TargetByName, solver...), (locator.TargetById, ...)]
```
I can think of a few cleaner solutions to this so that `register_nested_ui_solvers` could still be used, but am unsure what would be best.

Reply:
Offline discussion: I didn't think this through. The `solvers` list as a class variable does not allow us to reuse refactored functions that register many things. The current approach with extending the method in the subclass is correct.